quickwit-storage/src/bundle_storage.rs

```diff
@@ -0,0 +1,282 @@
+/*
+    Quickwit
+    Copyright (C) 2021 Quickwit Inc.
+
+    Quickwit is offered under the AGPL v3.0 and as commercial software.
+    For commercial licensing, contact us at hello@quickwit.io.
+
+    AGPL:
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use crate::Storage;
+use crate::{StorageError, StorageResult};
+use async_trait::async_trait;
+use bytes::Bytes;
+use quickwit_common::HOTCACHE_FILENAME;
+use serde::Deserialize;
+use serde::Serialize;
+use std::collections::HashMap;
+use std::fmt;
+use std::fmt::Debug;
+use std::fs::{File, OpenOptions};
+use std::io::{self, ErrorKind, Write};
+use std::ops::Range;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use tantivy::common::BinarySerializable;
+use tantivy::HasLen;
+use tracing::error;
+
+/// Filename used for the bundle.
+pub const BUNDLE_FILENAME: &str = "bundle";
```

Comment:
We'll probably drop the concept of split directory now that everything is a file. It will make operations nicer.
(one will be able to ls the index directory and see what is in there)

Reply:
what are the implications of dropping the concept of split directory?

When we `ls` the directory we just see the bundle file, or do you mean a virtual ls on the bundle metadata?